app/workers/functions.php

```diff
@@ -412,48 +371,62 @@ public function execute(string $trigger, string $projectId, string $executionId,
             $cpus = App::getEnv('_APP_FUNCTIONS_CPUS', '');
             $memory = App::getEnv('_APP_FUNCTIONS_MEMORY', '');
             $swap = App::getEnv('_APP_FUNCTIONS_MEMORY_SWAP', '');
-            $exitCode = Console::execute("docker run ".
-                " -d".
-                " --entrypoint=\"\"".
-                (empty($cpus) ? "" : (" --cpus=".$cpus)).
-                (empty($memory) ? "" : (" --memory=".$memory."m")).
-                (empty($swap) ? "" : (" --memory-swap=".$swap."m")).
-                " --name={$container}".
-                " --label appwrite-type=function".
-                " --label appwrite-created={$executionTime}".
-                " --volume {$tagPathTargetDir}:/tmp:rw".
-                " --workdir /usr/local/src".
-                " ".\implode(" ", $vars).
-                " {$runtime['image']}".
-                " tail -f /dev/null"
-            , '', $stdout, $stderr, 30);
-
-            if($exitCode !== 0) {
-                throw new Exception('Failed to create function environment: '.$stderr);
+
+            $orchestration->setCpus($cpus);
+            $orchestration->setMemory($memory);
+            $orchestration->setSwap($swap);
+
+            foreach($vars as $key => $value) {
+                $value = strval($value);
             }
```

Comment:
Do we need to pass `$value` by reference to mutate the original array items? Not sure how PHP handles this.